src/components/input-dropdown/input-dropdown.component.spec.ts

```diff
@@ -1,77 +1,200 @@
+    let fixture: ComponentFixture<InputDropdownComponent<any>>;
+    let overlayContainer: OverlayContainer;
+    let overlayContainerElement: HTMLElement;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [
+                InputDropdownModule,
+                NoopAnimationsModule
+            ]
+        }).overrideComponent(InputDropdownComponent, {
+            set: { changeDetection: ChangeDetectionStrategy.Default }
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(InputDropdownComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should reset filter correctly', () => {
+        component._filterText = 'Tralala';
+
+        spyOn(component.filterChange, 'emit');
+        spyOn(component.filterInputElement.nativeElement, 'focus');
+
+        component.resetFilter(new MouseEvent('click'));
+
+        fixture.detectChanges();
+        expect(component._filterText).toEqual('');
+        expect(component.filterChange.emit).toHaveBeenCalledWith('');
+        expect(component.filterInputElement.nativeElement.focus).toHaveBeenCalledWith();
+    });
+
+    it('should write value', () => {
+        spyOn(component.selectedChange, 'emit');
+        const newValue = 'Bla';
+
+        component.writeValue(newValue);
+
+        expect(component.selected).toEqual(newValue);
+    });
+
+    it('should register onChange', () => {
+        const callbackObject = { callback: () => true };
+        spyOn(callbackObject, 'callback');
+        const newValue = 'Blubb';
+
+        component.registerOnChange(callbackObject.callback);
+        component.onChange(newValue);
+
+        expect(callbackObject.callback).toHaveBeenCalledWith(newValue);
+    });
+
+    it('should register onTouched', () => {
+        const callbackObject = { callback: () => true };
+        spyOn(callbackObject, 'callback');
+
+        component.registerOnTouched(callbackObject.callback);
+        component.onTouched();
+
+        expect(callbackObject.callback).toHaveBeenCalledWith();
+    });
+});
+
+@Component({
+    selector: 'app-dropdown-test',
+    template: `
+        <ux-input-dropdown
+            [dropdownOpen]="dropdownOpen"
+            (dropdownOpenChange)="dropdownOpenChange($event)">
+
+            <ng-template #displayContent>
+                <b>Selection:</b> {{ selected ? selected.name : '(none)' }}
+            </ng-template>
+
+            <div class="radio-button-container"
+                 uxTabbableList>
+                <ux-radio-button
+                    uxTabbableListItem
+                    *ngFor="let option of filteredOptionList">
+                </ux-radio-button>
+            </div>
+        </ux-input-dropdown>
+    `
+})
+export class InputDropdownTestComponent {
+
+    inputFocusHandler() {
```

Comment:
Is this called? This doesn't seem right, this looks like we are trying to fake what the component does when it is focused? If so we wouldn't want to test against that as we are not actually testing the component, if the component focus behaviour stopped working the test would still pass which makes the test useless.

If you want to test what happens when the input is focused, find the input element using `nativeElement.querySelector` and call the `focus` function on the HTMLElemenet object